packages/rest/__tests__/client/AbstractRestClient.test.ts

```diff
@@ -640,6 +641,78 @@ describe("AbstractRestClient tests", () => {
         expect(httpsRequestFnc).toBeCalled();
     });
 
+    it("should create buildOptions according to input parameter options 2", async () => {
+
+        const httpsRequestFnc = jest.fn((options, callback) => {
+            expect(options).toMatchSnapshot();
+            const emitter = new MockHttpRequestResponse();
+            ProcessUtils.nextTick(() => {
+                callback(new EventEmitter());
+                ProcessUtils.nextTick(() => {
+                    emitter.emit("error", "value");
+                });
+            });
+            return emitter;
+        });
+
+        (https.request as any) = httpsRequestFnc;
+
+        let error;
+        try {
+            await RestClient.getExpectString(
+                new Session({
+                    hostname: "test",
+                    port: 8080,
+                    protocol: "https",
+                    basePath: "baseURL",
+                    type: "cert-pem",
+                    cert: join(__dirname, "..", "..", "..", "..", "__tests__", "__integration__",
+                        "cmd", "__tests__", "integration", "cli", "auth", "__resources__", "fakeCert.cert"),
+                    certKey: join(__dirname, "..", "..", "..", "..", "__tests__", "__integration__",
+                        "cmd", "__tests__", "integration", "cli", "auth", "__resources__", "fakeKey.key"),
+                }),
+                "/resource");
+        } catch (thrownError) {
+            error = thrownError;
+        }
+        expect(httpsRequestFnc).toBeCalled();
```

Comment:
Should there be another `expect` to check that `error` is undefined?

Reply:
Yep, you're right. Added.